storage_image_product/models/image_tag.py

```diff
@@ -22,7 +24,26 @@ def _get_default_apply_on(self):
         )
 
     name = fields.Char()
+    tech_name = fields.Char(string="Technical name")
     apply_on = fields.Selection(
         selection=[("product", "Product"), ("category", "Category")],
         default=lambda self: self._get_default_apply_on(),
     )
+
+    @api.model
+    def create(self, vals):
+        self._handle_tech_name(vals)
+        return super().create(vals)
+
+    def write(self, vals):
+        self._handle_tech_name(vals)
+        return super().write(vals)
+
+    def _handle_tech_name(self, vals):
```

Comment:
hmm this code looks familiar :thinking: :smile: 

@JordiBForgeFlow after re-implementing it here and there, in the ctx of server env I've added [this mixin](https://github.com/OCA/server-env/blob/13.0/server_environment/models/server_env_tech_name_mixin.py) and removed duplicated code from shopinvader and search engine modules.
Maybe it's time to make it a generic mixin?

Reply:
But then you need to add an extra dependency, just for this, right? IMHO in this case it's not really justified. I just used that because it was nice. I mean, if this feature was provided by a pip module, then I would agree to reuse.

Reply:
storage_backend already depends on server env https://github.com/OCA/storage/blob/13.0/storage_backend/__manifest__.py#L15

The mixin can be kept there. Also, this feature is something that it should be required for product attributes as well because they don't have unique key and the filter names in search engine keys may vary from lang to lang.

Reply:
BTW you can already use PIP to install modules :wink: